internal/repository/indexmap_test.go

```diff
@@ -44,26 +44,38 @@ func TestIndexMapForeach(t *testing.T) {
 		m.add(id, i, uint32(i), uint32(i))
 	}
 
-	seen := make(map[int]struct{})
-	m.foreach(func(e *indexEntry) bool {
-		i := int(e.id[0])
-		rtest.Assert(t, i < N, "unknown id %v in indexMap", e.id)
-		rtest.Equals(t, i, e.packIndex)
-		rtest.Equals(t, i, int(e.length))
-		rtest.Equals(t, i, int(e.offset))
-
-		seen[i] = struct{}{}
-		return true
-	})
-
-	rtest.Equals(t, N, len(seen))
+	testForEach := func() {
+		seen := make(map[int]struct{})
+		m.foreach(func(e *indexEntry) bool {
+			i := int(e.id[0])
+			rtest.Assert(t, i < N, "unknown id %v in indexMap", e.id)
+			rtest.Equals(t, i, e.packIndex)
+			rtest.Equals(t, i, int(e.length))
+			rtest.Equals(t, i, int(e.offset))
+
+			seen[i] = struct{}{}
```

Comment:
This doesn't count duplicate ids. Please make sure that your tests actually _fail_ without the fixed code. So either write the test first or undo the change afterwards and check that the test fails.

Reply:
My bad :-1: 
I'll change this.